src/leiningen/monolith.clj

```diff
@@ -274,28 +275,68 @@
     (fingerprint/clear project opts more)))
 
 
+(defn ^:higher-order with-dependency-set
+  "Run a task with a set of managed dependencies from a named dependency set.
+
+   Overrides the dependencies from the named dependency set into the project.
+   For the root project, this means the managed dependencies will be overwritten
+   with the dependencies from the named set. For subprojects, the
+   `:monolith/dependency-set` metadata key will be set to the named set. If the
+   project does not have the `:monolith/dependency-set` key defined, it will be
+   added.
+
+   For combining with other higher-order tasks, this task should be the last
+   evaluated.
```

Comment:
Some clarification on why this is would be useful. I'm assuming you're talking about a case like `lein monolith with-dependency-set :foo monolith each ...`?

Reply:
Correct, the behavior is a bit non-deterministic if you do `lein monolith with-dependency-set :foo monolith each ...` but from my testing looks to work as expected the other way around. I can clarify in the doc here what "should be the last evaluated" means.

Reply:
Aborting makes sense to me, better to guide the user in the right direction loudly than through docs.